nixos/modules/services/logging/filebeat.nix

```
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.filebeat;
+
+  filebeatYml = pkgs.writeText "filebeat.yml" ''
+    ${cfg.extraConfig}
+  '';
+
+  # This is extremely ugly.
+  # Using `.out` here pulls in all source code of all recursive go
+  # dependencies of filebeat.
+  # We do this so that filebeat can find `filebeat.template.json`
+  # (and other files that for the Debian bindist would be in /etc/filebeat).
+  # Unfortunately it's not clear how we can add them to the `buildGoPackage`
+  # that creates `pkgs.filebeat`, and it is also not clear how we can
+  # determine from the filebeat source code which files we have to copy.
+  configDir = "${pkgs.filebeat.out}/share/go/src/github.com/elastic/beats/filebeat";
+in
+{
+  options = {
+
+    services.filebeat = {
+
+      enable = mkEnableOption "filebeat";
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "/var/lib/filebeat";
+        description = "The state directory. filebeat's own logs and other data are stored here.";
+      };
+
+      extraConfig = mkOption {
```

```suggestion
      settings = mkOption {
```

This is commonly called settings now, see rfc 42